### ISSUE TITLE
Exp7: Batch size scheduling

### ISSUE THREAD

**Issue by Micky774** [quinngroup/CiliaRepresentation#10]
Experiment with initial batch sizes and batch size scheduling

**Comment by Micky774** [comment]
As per Exp2 analysis, when designing this experiment, at least for the initial implementation bear in mind that we want to look at behavior over time, not over epoch.

**Comment by Micky774** [comment]
As an update, this issue will particularly focus on accumulating gradients which we'll refer to as _rolling_. That is to say, we should add an argument `--roll k` which allows us to specify a factor k by which to roll/accumulate the gradients. So gradients are accumulated and only backpropogated/zeroed every k batches. This results in an effective batch size of k*N*M where N is the entered per-card batch-size, M is the number of cards being used, and k is the roll factor.

As an example, running a per-card batch-size of 25 with a roll factor of 3 while training on 2 cards would result in an effective batch-size of 150.
